src/components/ModalDialog/ModalDialogComponent.tsx

```diff
@@ -89,8 +99,14 @@ export const ModalDialogComponent = () => {
         { headers: { 'Content-Type': 'application/json' } }
       )
       .then((response) => {
-        const addedNode = [{ ...response.data[0], x, y }]
-        manualAddNode(addedNode, nodeLabels, dispatch);
+        if (isNodeDialog) {
+          const addedNode = [{ ...response.data[0], x, y }];
+          manualAddNode(addedNode, nodeLabels, dispatch);
+        }
+        else {
+          const addedEdge = response.data;
+          manualAddEdge(addedEdge, nodeLabels, dispatch);
+        }

```

Comment:
Try to not repeat function calls in each branch of an if-else. Set the variable in the if-else, and call the function once outside the if-else. This helps make more maintainable code if you need to refactor the function later.
e.g.
```
const add = isNodeDialog ? [{ ...response.data[0], x, y }] : response.data;
manualAddNode(add, nodeLabels, dispatch);
```